pkg/rpcservice/listen.go

```diff
@@ -0,0 +1,200 @@
+package rpcservice
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/stripe/stripe-cli/pkg/proxy"
+	"github.com/stripe/stripe-cli/pkg/websocket"
+	"github.com/stripe/stripe-cli/rpc"
+)
+
+const webhooksWebSocketFeature = "webhooks"
+
+// IProxy enables mocking a proxy object in tests
+type IProxy interface {
+	Run(context.Context) error
+}
+
+var createProxy = func(cfg *proxy.Config) (IProxy, error) {
+	return proxy.Init(cfg)
+}
+
+// Listen returns a stream of webhook events and forwards them to a local endpoint
+func (srv *RPCService) Listen(req *rpc.ListenRequest, stream rpc.StripeCLI_ListenServer) error {
+	deviceName, err := srv.cfg.UserCfg.Profile.GetDeviceName()
+	if err != nil {
+		return err
+	}
+
+	key, err := srv.cfg.UserCfg.Profile.GetAPIKey(req.Live)
+	if err != nil {
+		return err
+	}
+
+	// validate forward-urls args
+	if req.UseConfiguredWebhooks && len(req.ForwardTo) > 0 {
+		if strings.HasPrefix(req.ForwardTo, "/") {
+			return status.Error(codes.InvalidArgument, "forward_to cannot be a relative path when loading webhook endpoints from the API")
+		}
+		if strings.HasPrefix(req.ForwardConnectTo, "/") {
+			return status.Error(codes.InvalidArgument, "forward_connect_to cannot be a relative path when loading webhook endpoints from the API")
+		}
+	} else if req.UseConfiguredWebhooks && len(req.ForwardTo) == 0 {
+		return status.Error(codes.InvalidArgument, "load_from_webhooks_api requires a location to forward to with forward_to")
+	}
+
+	logger := log.StandardLogger()
+	proxyVisitor := createProxyVisitor(&stream)
+	proxyOutCh := make(chan websocket.IElement)
+
+	p, err := createProxy(&proxy.Config{
```

Comment:
this is where an err should be re turned if one of the arguments is invalid (cc previous comment on urls validation). 

If there are any challenges with having the url validation extracted from here, I'd at least make it happen inside `createProxy`